### PR TITLE
CM-20549 Cycode CLI - dev mode via sending request to localhost with fixing routing

### PR DESCRIPTION
**Pull request by morsa4406**

### CODE REVIEW

File: cyclient/cycode_client_base.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        )
+        response.raise_for_status()
+        return response
+
+    def get_request_headers(self, additional_headers: dict = None):
+        if additional_headers is None:
+            return self.MANDATORY_HEADERS
+        return {**self.MANDATORY_HEADERS, **additional_headers}
+
+    def append_to_endpoind(self):
```
Comment (MichalBor) [inline cyclient/cycode_client_base.py:65]: change to build_full_url and pass here he url and endpoint

File: cyclient/cycode_dev_based_client.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    def __init__(self, api_url):
+        super().__init__(api_url)
+
+    def get_request_headers(self, additional_headers: dict = None):
+        headers = super().get_request_headers(additional_headers=additional_headers)
+        headers['X-Tenant-Id'] = dev_tenant_id
+
+        return headers
+
+    def append_to_endpoind(self):
```
Comment (MichalBor) [inline cyclient/cycode_dev_based_client.py:20]: same

File: cyclient/scan_client.py [hunk 3]
```diff
@@ -1,86 +1,86 @@
...
-            response = self.cycode_client.post(url_path=url_path, files=files)
+            response = self.scan_cycode_client.post(url_path=url_path, files=files)
             return self.parse_scan_response(response)
         except Exception as e:
             self._handle_exception(e)
 
     def zipped_file_scan(self, scan_type: str, zip_file: InMemoryZip, scan_id: str, scan_parameters: dict,
                          is_git_diff: bool = False) -> models.ZippedFileScanResult:
-        url_path = f"{self.get_service_name(scan_type)}/{self.SCAN_CONTROLLER_PATH}/zipped-file"
+        url_path = self.scan_config.get_zipped_file_scan_controller_path(scan_type)
```
Comment (MichalBor) [inline cyclient/scan_client.py:40]: get from the config the path to the controller and add the route and relevant query params here (relevant to all the requests here)

**Review by MichalBor — CHANGES_REQUESTED** [review]
well done! great progress :)

File: cyclient/cycode_client_base.py (hunk 1 above)
Comment (morsa4406) [reply cyclient/cycode_client_base.py]: Fixed

File: cyclient/cycode_dev_based_client.py (hunk 2 above)
Comment (morsa4406) [reply cyclient/cycode_dev_based_client.py]: Fixed

File: cyclient/scan_client.py (hunk 3 above)
Comment (morsa4406) [reply cyclient/scan_client.py]: Fixed

**Review by MichalBor — APPROVED** [review]

**Pull request merged**